android/src/com/frostwire/android/gui/adapters/menu/RenameFileMenuAction.java

```diff
@@ -133,12 +145,14 @@ protected void initComponents(Dialog dlg, Bundle savedInstanceState) {
             noButton.setOnClickListener(new ButtonOnClickListener(this, false));
         }
 
-        public void rename() {
+        public String rename() {
             EditText input = findView(getDialog(), R.id.dialog_default_input_text);
+            String newFileName = new String(input.getText().toString());
```

Comment:
this does nothing in terms of reference or memory leak. This constructor just re-pin the internal memory of the original string. There are only a handful of justified cases for which you want to actually copy the chars of another string and all of them include a proper subset.